theano/sandbox/cuda/dnn.py

```diff
@@ -443,8 +443,8 @@ def grad(self, inp, grads):
 
         top = gpu_contiguous(top)
 
-        d_img = GpuDnnConvGradI()(kerns, top, img.zeros_like(), desc)
-        d_kerns = GpuDnnConvGradW()(img, top, kerns.zeros_like(), desc)
+        d_img = GpuDnnConvGradI()(kerns, top, img, desc)
+        d_kerns = GpuDnnConvGradW()(img, top, kerns, desc)
```

Comment:
After discussion, we determined that it would be better to use the new GpuAllocEmpty for this.  Otherwise it may force the computation of these inputs even if we only need their shape.  This is not a problem for microbenchmarks, but could affect cases like deconvolution nets.